src/universal/modules/meeting/containers/MeetingContainer/MeetingContainer.js

```diff
@@ -149,11 +153,36 @@ export default class MeetingContainer extends Component {
 
   shouldComponentUpdate(nextProps) {
     const {localPhaseItem, router, params: {localPhase}, team} = nextProps;
-    const {team: oldTeam} = this.props;
-
-    // only needs to run when the url changes or the team subscription initializes
-    // make sure the url is legit, but only run once (when the initial team subscription comes back)
-    return handleRedirects(team, localPhase, localPhaseItem, oldTeam, router);
+    const {dispatch, isFacilitating, team: oldTeam} = this.props;
+    const safeRoute = handleRedirects(team, localPhase, localPhaseItem, oldTeam, router);
+    if (safeRoute) {
+      return true;
+    }
+    // if we call router.push
+    if (Date.now() - infiniteLoopTimer < 1000) {
+      if (++infiniteloopCounter >= 10) {
+        // if we're changing locations 10 times in a second, it's probably infinite
+        if (isFacilitating) {
+          const variables = {
+            teamId: team.id,
+            nextPhase: CHECKIN,
+            nextPhaseItem: 1,
+            force: true
+          };
+          cashay.mutate('moveMeeting', {variables});
+        }
+        this.gotoItem(1, CHECKIN);
+        dispatch(showError({
```

Comment:
We can _positively_ use sentry.io here to dispatch an automated report. We could change the message to "we've automatically filed this incident with our team," or some such. I'll open an issue for us to track.